### ISSUE TITLE
Qt6 support

### ISSUE THREAD

**Issue by sketch34** [KDAB/GammaRay#631]
I'm running GammaRay built off the `work/qt6-support` branch. I get this output with both **Qt 6.0.1** and **Qt 6.1.0**:

```
gammaray_probe_inject()
Trying to construct an instance of an invalid type, type id: 27
Trying to construct an instance of an invalid type, type id: 59
Trying to construct an instance of an invalid type, type id: 60
Trying to construct an instance of an invalid type, type id: 61
...
```

Does GammaRay not currently support Qt6?

Built on Windows 10 with MSVC 2019
cmake version 3.19.6

**Comment by Waqar144** [comment]
@sketch34 can you try running the same branch again and see if things are better. A lot more work was done to improve Qt6 support.

**Comment by JohannesKauffmann** [comment]
I'm on Windows 10, building with Qt 6.2.2, MinGW 9.0.0 (GCC 11.2.0) and CMake 3.21.1.

Currently, I get a lot of erros when building the `ui` subproject:

```
19:51:08: Running steps for project GammaRay...
19:51:08: Starting: "C:\Qt\Tools\CMake_64\bin\cmake.exe" --build C:/build-GammaRay-Desktop_Qt_6_2_2_MinGW_64_bit-Debug --target gammaray_ui_internal
[1/112 0.7/sec] Building CXX object common/CMakeFiles/gammaray_common.dir/remoteviewinterface.cpp.obj
FAILED: common/CMakeFiles/gammaray_common.dir/remoteviewinterface.cpp.obj 
C:\Qt\Tools\mingw900_64\bin\g++.exe -DMINGW_HAS_SECURE_API=1 -DQT_CORE_LIB -DQT_DISABLE_DEPRECATED_BEFORE=0x050500 -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_NO_CAST_TO_ASCII -DQT_NO_URL_CAST_FROM_STRING -DQT_USE_FAST_CONCATENATION -DQT_USE_FAST_OPERATOR_PLUS -DUNICODE -DWIN32 -DWIN64 -D_ENABLE_EXTENDED_ALIGNED_STORAGE -D_UNICODE -D_USING_V110_SDK71_=1 -D_WIN64 -Dgammaray_common_EXPORTS -IC:/build-GammaRay-Desktop_Qt_6_2_2_MinGW_64_bit-Debug/common -IC:/GammaRay/common -IC:/build-GammaRay-Desktop_Qt_6_2_2_MinGW_64_bit-Debug/common/gammaray_common_autogen/include -IC:/GammaRay -IC:/GammaRay/3rdparty -IC:/build-GammaRay-Desktop_Qt_6_2_2_MinGW_64_bit-Debug -isystem C:/Qt/6.2.2/mingw_64/include/QtCore -isystem C:/Qt/6.2.2/mingw_64/include -isystem C:/Qt/6.2.2/mingw_64/mkspecs/win32-g++ -isystem C:/Qt/6.2.2/mingw_64/include/QtNetwork -isystem C:/Qt/6.2.2/mingw_64/include/QtGui -Wunused-but-set-variable -Wlogical-op -Wsizeof-pointer-memaccess -Wreorder -Wall -Wextra -pedantic -Woverloaded-virtual -Winit-self -Wmissing-include-dirs -Wunused -Wundef -Wpointer-arith -Wmissing-noreturn -Werror=return-type -Wswitch -std=gnu++0x -fpermissive -fno-operator-names -Wzero-as-null-pointer-constant -g -fvisibility=hidden -fno-keep-inline-dllexport -std=gnu++17 -MD -MT common/CMakeFiles/gammaray_common.dir/remoteviewinterface.cpp.obj -MF common\CMakeFiles\gammaray_common.dir\remoteviewinterface.cpp.obj.d -o common/CMakeFiles/gammaray_common.dir/remoteviewinterface.cpp.obj -c C:/GammaRay/common/remoteviewinterface.cpp
C:/GammaRay/common/remoteviewinterface.cpp: In function 'QDataStream& operator<<(QDataStream&, const QList<QEventPoint>&)':
C:/GammaRay/common/remoteviewinterface.cpp:75:16: error: 'const class QEventPoint' has no member named 'rect'
   75 |         s << p.rect() << p.sceneRect() << p.screenRect();
      |                ^~~~
C:/GammaRay/common/remoteviewinterface.cpp:75:28: error: 'const class QEventPoint' has no member named 'sceneRect'
   75 |         s << p.rect() << p.sceneRect() << p.screenRect();
      |                            ^~~~~~~~~
C:/GammaRay/common/remoteviewinterface.cpp:75:45: error: 'const class QEventPoint' has no member named 'screenRect'
   75 |         s << p.rect() << p.sceneRect() << p.screenRect();
      |                                             ^~~~~~~~~~
C:/GammaRay/common/remoteviewinterface.cpp:81:16: error: 'const class QEventPoint' has no member named 'flags'
   81 |         s << p.flags();
      |                ^~~~~
C:/GammaRay/common/remoteviewinterface.cpp:82:16: error: 'const class QEventPoint' has no member named 'rawScreenPositions'; did you mean 'scenePosition'?
   82 |         s << p.rawScreenPositions();
      |                ^~~~~~~~~~~~~~~~~~
      |                scenePosition
C:/GammaRay/common/remoteviewinterface.cpp: In function 'QDataStream& operator>>(QDataStream&, QList<QEventPoint>&)':
C:/GammaRay/common/remoteviewinterface.cpp:105:55: error: 'setId' is not a member of 'QTouchEvent::TouchPoint' {aka 'QEventPoint'}
  105 |         setPointValue(s, p, &QTouchEvent::TouchPoint::setId);
      |                                                       ^~~~~
C:/GammaRay/common/remoteviewinterface.cpp:106:55: error: 'setState' is not a member of 'QTouchEvent::TouchPoint' {aka 'QEventPoint'}
  106 |         setPointValue(s, p, &QTouchEvent::TouchPoint::setState);
      |                                                       ^~~~~~~~
C:/GammaRay/common/remoteviewinterface.cpp:108:55: error: 'setRect' is not a member of 'QTouchEvent::TouchPoint' {aka 'QEventPoint'}
  108 |         setPointValue(s, p, &QTouchEvent::TouchPoint::setRect);
      |                                                       ^~~~~~~
C:/GammaRay/common/remoteviewinterface.cpp:109:55: error: 'setSceneRect' is not a member of 'QTouchEvent::TouchPoint' {aka 'QEventPoint'}
  109 |         setPointValue(s, p, &QTouchEvent::TouchPoint::setSceneRect);
      |                                                       ^~~~~~~~~~~~
C:/GammaRay/common/remoteviewinterface.cpp:110:55: error: 'setScreenRect' is not a member of 'QTouchEvent::TouchPoint' {aka 'QEventPoint'}
  110 |         setPointValue(s, p, &QTouchEvent::TouchPoint::setScreenRect);
      |                                                       ^~~~~~~~~~~~~
C:/GammaRay/common/remoteviewinterface.cpp:112:55: error: 'setNormalizedPos' is not a member of 'QTouchEvent::TouchPoint' {aka 'QEventPoint'}
  112 |         setPointValue(s, p, &QTouchEvent::TouchPoint::setNormalizedPos);
      |                                                       ^~~~~~~~~~~~~~~~
C:/GammaRay/common/remoteviewinterface.cpp:114:55: error: 'setStartPos' is not a member of 'QTouchEvent::TouchPoint' {aka 'QEventPoint'}
  114 |         setPointValue(s, p, &QTouchEvent::TouchPoint::setStartPos);
      |                                                       ^~~~~~~~~~~
C:/GammaRay/common/remoteviewinterface.cpp:115:55: error: 'setStartScenePos' is not a member of 'QTouchEvent::TouchPoint' {aka 'QEventPoint'}
  115 |         setPointValue(s, p, &QTouchEvent::TouchPoint::setStartScenePos);
      |                                                       ^~~~~~~~~~~~~~~~
C:/GammaRay/common/remoteviewinterface.cpp:116:55: error: 'setStartScreenPos' is not a member of 'QTouchEvent::TouchPoint' {aka 'QEventPoint'}
  116 |         setPointValue(s, p, &QTouchEvent::TouchPoint::setStartScreenPos);
      |                                                       ^~~~~~~~~~~~~~~~~
C:/GammaRay/common/remoteviewinterface.cpp:117:55: error: 'setStartNormalizedPos' is not a member of 'QTouchEvent::TouchPoint' {aka 'QEventPoint'}
  117 |         setPointValue(s, p, &QTouchEvent::TouchPoint::setStartNormalizedPos);
      |                                                       ^~~~~~~~~~~~~~~~~~~~~
C:/GammaRay/common/remoteviewinterface.cpp:119:55: error: 'setLastPos' is not a member of 'QTouchEvent::TouchPoint' {aka 'QEventPoint'}
  119 |         setPointValue(s, p, &QTouchEvent::TouchPoint::setLastPos);
      |                                                       ^~~~~~~~~~
C:/GammaRay/common/remoteviewinterface.cpp:120:55: error: 'setLastScenePos' is not a member of 'QTouchEvent::TouchPoint' {aka 'QEventPoint'}
  120 |         setPointValue(s, p, &QTouchEvent::TouchPoint::setLastScenePos);
      |                                                       ^~~~~~~~~~~~~~~
C:/GammaRay/common/remoteviewinterface.cpp:121:55: error: 'setLastScreenPos' is not a member of 'QTouchEvent::TouchPoint' {aka 'QEventPoint'}
  121 |         setPointValue(s, p, &QTouchEvent::TouchPoint::setLastScreenPos);
      |                                                       ^~~~~~~~~~~~~~~~
C:/GammaRay/common/remoteviewinterface.cpp:122:55: error: 'setLastNormalizedPos' is not a member of 'QTouchEvent::TouchPoint' {aka 'QEventPoint'}
  122 |         setPointValue(s, p, &QTouchEvent::TouchPoint::setLastNormalizedPos);
      |                                                       ^~~~~~~~~~~~~~~~~~~~
C:/GammaRay/common/remoteviewinterface.cpp:124:55: error: 'setPressure' is not a member of 'QTouchEvent::TouchPoint' {aka 'QEventPoint'}
  124 |         setPointValue(s, p, &QTouchEvent::TouchPoint::setPressure);
      |                                                       ^~~~~~~~~~~
C:/GammaRay/common/remoteviewinterface.cpp:125:55: error: 'setVelocity' is not a member of 'QTouchEvent::TouchPoint' {aka 'QEventPoint'}
  125 |         setPointValue(s, p, &QTouchEvent::TouchPoint::setVelocity);
      |                                                       ^~~~~~~~~~~
C:/GammaRay/common/remoteviewinterface.cpp:127:55: error: 'setFlags' is not a member of 'QTouchEvent::TouchPoint' {aka 'QEventPoint'}
  127 |         setPointValue(s, p, &QTouchEvent::TouchPoint::setFlags);
      |                                                       ^~~~~~~~
C:/GammaRay/common/remoteviewinterface.cpp:128:55: error: 'setRawScreenPositions' is not a member of 'QTouchEvent::TouchPoint' {aka 'QEventPoint'}
  128 |         setPointValue(s, p, &QTouchEvent::TouchPoint::setRawScreenPositions);
      |                                                       ^~~~~~~~~~~~~~~~~~~~~
ninja: build stopped: subcommand failed.
19:51:10: The process "C:\Qt\Tools\CMake_64\bin\cmake.exe" exited with code 1.
Error while building/deploying project GammaRay (kit: Desktop Qt 6.2.2 MinGW 64-bit)
When executing step "Build"
19:51:10: Elapsed time: 00:02.
```

**Comment by Waqar144** [comment]
please pass `-DGAMMARAY_QT6_BUILD=ON` to cmake to compile with Qt6

**Comment by JohannesKauffmann** [comment]
Hi, sorry for that, I missed that in the CMake options.

I tried to further minimize the build for testing purposes, with the following options:

```
-DGAMMARAY_QT6_BUILD:BOOL=ON -DBACKWARD_TESTS:BOOL=OFF -DBUILD_TESTING:BOOL=OFF -DGAMMARAY_BUILD_DOCS:BOOL=OFF -DGAMMARAY_DISABLE_FEEDBACK:BOOL=ON
```

So with the options above (which includes the Qt6 CMake option), it still won't build, but there are a lot less errors:

```
[274/597 0.8/sec] Building CXX object plugins/quickinspector/CMakeFiles/gammaray_quickinspector.dir/materialextension/qquickopenglshadereffectmaterialadaptor.cpp.obj
FAILED: plugins/quickinspector/CMakeFiles/gammaray_quickinspector.dir/materialextension/qquickopenglshadereffectmaterialadaptor.cpp.obj 
C:\Qt\Tools\mingw900_64\bin\c++.exe -DMINGW_HAS_SECURE_API=1 -DQT_CORE_LIB -DQT_DISABLE_DEPRECATED_BEFORE=0x050500 -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_NO_CAST_TO_ASCII -DQT_NO_URL_CAST_FROM_STRING -DQT_OPENGL_LIB -DQT_QMLMODELS_LIB -DQT_QML_LIB -DQT_QUICK_LIB -DQT_USE_FAST_CONCATENATION -DQT_USE_FAST_OPERATOR_PLUS -DUNICODE -DWIN32 -DWIN64 -D_ENABLE_EXTENDED_ALIGNED_STORAGE -D_UNICODE -D_USING_V110_SDK71_=1 -D_WIN64 -Dgammaray_quickinspector_EXPORTS -IC:/build-GammaRay-Desktop_Qt_6_2_2_MinGW_64_bit-Release/Debug/plugins/quickinspector -IC:/GammaRay/plugins/quickinspector -IC:/build-GammaRay-Desktop_Qt_6_2_2_MinGW_64_bit-Release/Debug/plugins/quickinspector/gammaray_quickinspector_autogen/include -IC:/GammaRay -IC:/GammaRay/3rdparty -IC:/build-GammaRay-Desktop_Qt_6_2_2_MinGW_64_bit-Release/Debug -IC:/build-GammaRay-Desktop_Qt_6_2_2_MinGW_64_bit-Release/Debug/common -IC:/build-GammaRay-Desktop_Qt_6_2_2_MinGW_64_bit-Release/Debug/core -isystem C:/Qt/6.2.2/mingw_64/include/QtCore -isystem C:/Qt/6.2.2/mingw_64/include -isystem C:/Qt/6.2.2/mingw_64/mkspecs/win32-g++ -isystem C:/Qt/6.2.2/mingw_64/include/QtGui -isystem C:/Qt/6.2.2/mingw_64/include/QtQuick -isystem C:/Qt/6.2.2/mingw_64/include/QtQml -isystem C:/Qt/6.2.2/mingw_64/include/QtNetwork -isystem C:/Qt/6.2.2/mingw_64/include/QtQmlModels -isystem C:/Qt/6.2.2/mingw_64/include/QtOpenGL -isystem C:/Qt/6.2.2/mingw_64/include/QtQuick/6.2.2 -isystem C:/Qt/6.2.2/mingw_64/include/QtQuick/6.2.2/QtQuick -isystem C:/Qt/6.2.2/mingw_64/include/QtCore/6.2.2 -isystem C:/Qt/6.2.2/mingw_64/include/QtCore/6.2.2/QtCore -isystem C:/Qt/6.2.2/mingw_64/include/QtGui/6.2.2 -isystem C:/Qt/6.2.2/mingw_64/include/QtGui/6.2.2/QtGui -isystem C:/Qt/6.2.2/mingw_64/include/QtQmlModels/6.2.2 -isystem C:/Qt/6.2.2/mingw_64/include/QtQmlModels/6.2.2/QtQmlModels -isystem C:/Qt/6.2.2/mingw_64/include/QtQml/6.2.2 -isystem C:/Qt/6.2.2/mingw_64/include/QtQml/6.2.2/QtQml -Wunused-but-set-variable -Wlogical-op -Wsizeof-pointer-memaccess -Wreorder -Wall -Wextra -pedantic -Woverloaded-virtual -Winit-self -Wmissing-include-dirs -Wunused -Wundef -Wpointer-arith -Wmissing-noreturn -Werror=return-type -Wswitch -std=gnu++0x -fpermissive -fno-operator-names -Wzero-as-null-pointer-constant -g -fvisibility=hidden -fno-keep-inline-dllexport -std=gnu++17 -MD -MT plugins/quickinspector/CMakeFiles/gammaray_quickinspector.dir/materialextension/qquickopenglshadereffectmaterialadaptor.cpp.obj -MF plugins\quickinspector\CMakeFiles\gammaray_quickinspector.dir\materialextension\qquickopenglshadereffectmaterialadaptor.cpp.obj.d -o plugins/quickinspector/CMakeFiles/gammaray_quickinspector.dir/materialextension/qquickopenglshadereffectmaterialadaptor.cpp.obj -c C:/GammaRay/plugins/quickinspector/materialextension/qquickopenglshadereffectmaterialadaptor.cpp
C:/GammaRay/plugins/quickinspector/materialextension/qquickopenglshadereffectmaterialadaptor.cpp:35:10: fatal error: private/qquickopenglshadereffectnode_p.h: No such file or directory
   35 | #include <private/qquickopenglshadereffectnode_p.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
[276/597 0.8/sec] Building CXX object plugins/quickinspector/CMakeFiles/gammaray_quickinspector.dir/materialextension/materialextension.cpp.obj
FAILED: plugins/quickinspector/CMakeFiles/gammaray_quickinspector.dir/materialextension/materialextension.cpp.obj 
C:\Qt\Tools\mingw900_64\bin\c++.exe -DMINGW_HAS_SECURE_API=1 -DQT_CORE_LIB -DQT_DISABLE_DEPRECATED_BEFORE=0x050500 -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_NO_CAST_TO_ASCII -DQT_NO_URL_CAST_FROM_STRING -DQT_OPENGL_LIB -DQT_QMLMODELS_LIB -DQT_QML_LIB -DQT_QUICK_LIB -DQT_USE_FAST_CONCATENATION -DQT_USE_FAST_OPERATOR_PLUS -DUNICODE -DWIN32 -DWIN64 -D_ENABLE_EXTENDED_ALIGNED_STORAGE -D_UNICODE -D_USING_V110_SDK71_=1 -D_WIN64 -Dgammaray_quickinspector_EXPORTS -IC:/build-GammaRay-Desktop_Qt_6_2_2_MinGW_64_bit-Release/Debug/plugins/quickinspector -IC:/GammaRay/plugins/quickinspector -IC:/build-GammaRay-Desktop_Qt_6_2_2_MinGW_64_bit-Release/Debug/plugins/quickinspector/gammaray_quickinspector_autogen/include -IC:/GammaRay -IC:/GammaRay/3rdparty -IC:/build-GammaRay-Desktop_Qt_6_2_2_MinGW_64_bit-Release/Debug -IC:/build-GammaRay-Desktop_Qt_6_2_2_MinGW_64_bit-Release/Debug/common -IC:/build-GammaRay-Desktop_Qt_6_2_2_MinGW_64_bit-Release/Debug/core -isystem C:/Qt/6.2.2/mingw_64/include/QtCore -isystem C:/Qt/6.2.2/mingw_64/include -isystem C:/Qt/6.2.2/mingw_64/mkspecs/win32-g++ -isystem C:/Qt/6.2.2/mingw_64/include/QtGui -isystem C:/Qt/6.2.2/mingw_64/include/QtQuick -isystem C:/Qt/6.2.2/mingw_64/include/QtQml -isystem C:/Qt/6.2.2/mingw_64/include/QtNetwork -isystem C:/Qt/6.2.2/mingw_64/include/QtQmlModels -isystem C:/Qt/6.2.2/mingw_64/include/QtOpenGL -isystem C:/Qt/6.2.2/mingw_64/include/QtQuick/6.2.2 -isystem C:/Qt/6.2.2/mingw_64/include/QtQuick/6.2.2/QtQuick -isystem C:/Qt/6.2.2/mingw_64/include/QtCore/6.2.2 -isystem C:/Qt/6.2.2/mingw_64/include/QtCore/6.2.2/QtCore -isystem C:/Qt/6.2.2/mingw_64/include/QtGui/6.2.2 -isystem C:/Qt/6.2.2/mingw_64/include/QtGui/6.2.2/QtGui -isystem C:/Qt/6.2.2/mingw_64/include/QtQmlModels/6.2.2 -isystem C:/Qt/6.2.2/mingw_64/include/QtQmlModels/6.2.2/QtQmlModels -isystem C:/Qt/6.2.2/mingw_64/include/QtQml/6.2.2 -isystem C:/Qt/6.2.2/mingw_64/include/QtQml/6.2.2/QtQml -Wunused-but-set-variable -Wlogical-op -Wsizeof-pointer-memaccess -Wreorder -Wall -Wextra -pedantic -Woverloaded-virtual -Winit-self -Wmissing-include-dirs -Wunused -Wundef -Wpointer-arith -Wmissing-noreturn -Werror=return-type -Wswitch -std=gnu++0x -fpermissive -fno-operator-names -Wzero-as-null-pointer-constant -g -fvisibility=hidden -fno-keep-inline-dllexport -std=gnu++17 -MD -MT plugins/quickinspector/CMakeFiles/gammaray_quickinspector.dir/materialextension/materialextension.cpp.obj -MF plugins\quickinspector\CMakeFiles\gammaray_quickinspector.dir\materialextension\materialextension.cpp.obj.d -o plugins/quickinspector/CMakeFiles/gammaray_quickinspector.dir/materialextension/materialextension.cpp.obj -c C:/GammaRay/plugins/quickinspector/materialextension/materialextension.cpp
C:/GammaRay/plugins/quickinspector/materialextension/materialextension.cpp:52:10: fatal error: private/qquickopenglshadereffectnode_p.h: No such file or directory
   52 | #include <private/qquickopenglshadereffectnode_p.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
```

**Comment by Waqar144** [comment]
Hi, looking at your compile command, it seems like there is no `-DGAMMARAY_QT6_TODO` which would avoid the error you are facing.

I have pushed a small fix for this. Can you try again?

**Comment by JohannesKauffmann** [comment]
Looks like the fix now indeed defines `GAMMARAY_QT6_TODO`. However, unfortunately now I get the following error from `as.exe`:

```
[331/597] Building CXX object plugins/guisupport/CMakeFiles/gammaray_guisupport.dir/guisupport.cpp.obj
FAILED: plugins/guisupport/CMakeFiles/gammaray_guisupport.dir/guisupport.cpp.obj
C:\Qt\Tools\mingw900_64\bin\c++.exe -DGAMMARAY_QT6_TODO -DMINGW_HAS_SECURE_API=1 -DQT_CORE_LIB -DQT_DISABLE_DEPRECATED_BEFORE=0x050500 -DQT_GUI_LIB -DQT_NO_CAST_TO_ASCII -DQT_NO_URL_CAST_FROM_STRING -DQT_USE_FAST_CONCATENATION -DQT_USE_FAST_OPERATOR_PLUS -DUNICODE -DWIN32 -DWIN64 -D_ENABLE_EXTENDED_ALIGNED_STORAGE -D_UNICODE -D_USING_V110_SDK71_=1 -D_WIN64 -Dgammaray_guisupport_EXPORTS -IC:/build-GammaRay-Desktop_Qt_6_2_2_MinGW_64_bit-Release/Debug/plugins/guisupport -IC:/GammaRay/plugins/guisupport -IC:/build-GammaRay-Desktop_Qt_6_2_2_MinGW_64_bit-Release/Debug/plugins/guisupport/gammaray_guisupport_autogen/include -IC:/GammaRay -IC:/GammaRay/3rdparty -IC:/build-GammaRay-Desktop_Qt_6_2_2_MinGW_64_bit-Release/Debug -IC:/build-GammaRay-Desktop_Qt_6_2_2_MinGW_64_bit-Release/Debug/core -IC:/build-GammaRay-Desktop_Qt_6_2_2_MinGW_64_bit-Release/Debug/common -isystem C:/Qt/6.2.2/mingw_64/include/QtCore -isystem C:/Qt/6.2.2/mingw_64/include -isystem C:/Qt/6.2.2/mingw_64/mkspecs/win32-g++ -isystem C:/Qt/6.2.2/mingw_64/include/QtGui -isystem C:/Qt/6.2.2/mingw_64/include/QtGui/6.2.2 -isystem C:/Qt/6.2.2/mingw_64/include/QtGui/6.2.2/QtGui -isystem C:/Qt/6.2.2/mingw_64/include/QtCore/6.2.2 -isystem C:/Qt/6.2.2/mingw_64/include/QtCore/6.2.2/QtCore -Wunused-but-set-variable -Wlogical-op -Wsizeof-pointer-memaccess -Wreorder -Wall -Wextra -pedantic -Woverloaded-virtual -Winit-self -Wmissing-include-dirs -Wunused -Wundef -Wpointer-arith -Wmissing-noreturn -Werror=return-type -Wswitch -std=gnu++0x -fpermissive -fno-operator-names -Wzero-as-null-pointer-constant -g -fvisibility=hidden -fno-keep-inline-dllexport -std=gnu++17 -MD -MT plugins/guisupport/CMakeFiles/gammaray_guisupport.dir/guisupport.cpp.obj -MF plugins\guisupport\CMakeFiles\gammaray_guisupport.dir\guisupport.cpp.obj.d -o plugins/guisupport/CMakeFiles/gammaray_guisupport.dir/guisupport.cpp.obj -c C:/GammaRay/plugins/guisupport/guisupport.cpp
C:/GammaRay/plugins/guisupport/guisupport.cpp: In member function 'QIcon GammaRay::GuiSupport::createIcon(const QIcon&, QWindow*)':
C:/GammaRay/plugins/guisupport/guisupport.cpp:1195:57: warning: 'Qt::AA_UseHighDpiPixmaps' is deprecated: High-DPI pixmaps are always enabled. This attribute no longer has any effect. [-Wdeprecated-declarations]
 1195 |     const bool highDpiEnabled = qApp->testAttribute(Qt::AA_UseHighDpiPixmaps);
      |                                                         ^~~~~~~~~~~~~~~~~~~~
In file included from C:/Qt/6.2.2/mingw_64/include/QtCore/qbytearray.h:45,
                 from C:/Qt/6.2.2/mingw_64/include/QtCore/qmetatype.h:47,
                 from C:/Qt/6.2.2/mingw_64/include/QtCore/QMetaType:1,
                 from C:/GammaRay/common/sourcelocation.h:34,
                 from C:/GammaRay/core/probe.h:35,
                 from C:/GammaRay/core/toolfactory.h:42,
                 from C:/GammaRay/plugins/guisupport/guisupport.h:32,
                 from C:/GammaRay/plugins/guisupport/guisupport.cpp:29:
C:/Qt/6.2.2/mingw_64/include/QtCore/qnamespace.h:462:9: note: declared here
  462 |         AA_UseHighDpiPixmaps Q_DECL_ENUMERATOR_DEPRECATED_X(
      |         ^~~~~~~~~~~~~~~~~~~~
C:/Qt/Tools/mingw900_64/bin/../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../x86_64-w64-mingw32/bin/as.exe: plugins/guisupport/CMakeFiles/gammaray_guisupport.dir/guisupport.cpp.obj: too many sections (33086)
C:\Users\Johannes\AppData\Local\Temp\ccFXU4Df.s: Assembler messages:
C:\Users\Johannes\AppData\Local\Temp\ccFXU4Df.s: Fatal error: can't write 31 bytes to section .text of plugins/guisupport/CMakeFiles/gammaray_guisupport.dir/guisupport.cpp.obj: 'file too big'
C:/Qt/Tools/mingw900_64/bin/../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../x86_64-w64-mingw32/bin/as.exe: plugins/guisupport/CMakeFiles/gammaray_guisupport.dir/guisupport.cpp.obj: too many sections (33086)
C:\Users\Johannes\AppData\Local\Temp\ccFXU4Df.s: Fatal error: plugins/guisupport/CMakeFiles/gammaray_guisupport.dir/guisupport.cpp.obj: file too big
```

I'm not sure if that has something to do with Qt 6 support however.

So, additionally, I also tried to compile with MSVC instead of MinGW, against prebuild Qt 6.2.2 from the Qt Company, which succesfully passes the build.

**Comment by Waqar144** [comment]
Qt6 is now supported in master branch.
